projects/packages/identity-crisis/src/class-rest-endpoints.php

```diff
@@ -219,6 +237,31 @@ public static function fetch_url_secret() {
 		);
 	}
 
+	/**
+	 * Endpoint for comparing the existing secret.
+	 *
+	 * @param \WP_REST_Request $request The request sent to the WP REST API.
+	 *
+	 * @return WP_Error|\WP_REST_Response
+	 */
+	public static function compare_url_secret( $request ) {
+		$match = false;
+
+		$storage = new URL_Secret();
+
+		if ( $storage->exists() ) {
+			$remote_secret = $request->get_param( 'secret' );
+			$match         = $remote_secret && $storage->get_secret() === $remote_secret;
```

Comment:
Drive-by comment: Should we use [hash_equals](https://www.php.net/hash_equals) here?

Reply:
Good point, I'll use that 👍